cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/type/DatasetTypeManager.java

```diff
@@ -376,6 +378,27 @@ private void deployDefaultModules() {
     }
   }
 
+  private void deleteSystemModules() {
+    try {
+      mdsDatasets.execute(new TxCallable<MDSDatasets, Void>() {
+        @Override
+        public Void call(MDSDatasets context) throws Exception {
+          DatasetTypeMDS typeMDS = context.getTypeMDS();
+          Collection<DatasetModuleMeta> allDatasets = typeMDS.getModules(Constants.SYSTEM_NAMESPACE_ID);
+          for (DatasetModuleMeta ds : allDatasets) {
+            if (ds.getJarLocation() == null) {
```

Comment:
There is definitely overlap with the upgrade tool.  You're right that this change will not delete the old system modules due to the change in row keying.  And technically, the upgrade tool could skip the system datasets rows (both with this change and even without it), but I'm not sure it's worth the additional special casing in the upgrade tool itself.